pulser-core/pulser/sequence/sequence.py

```diff
@@ -671,6 +716,188 @@ def declare_variable(
             self._variables[name] = var
             return var
 
+    @seq_decorators.store
+    @seq_decorators.block_if_measured
+    def enable_eom_mode(
+        self,
+        channel: str,
+        amp_on: Union[float, Parametrized],
+        detuning_on: Union[float, Parametrized],
+        detuning_off_choice: DETUNING_OFF_OPTIONS = "closest-to-zero",
+    ) -> None:
+        """Puts a channel in EOM mode operation.
+
+        For channels with a finite modulation bandwidth and an EOM, operation
+        in EOM mode allows for the execution of square pulses with a higher
+        bandwidth than that which is tipically available. It can be turned on
+        and off through the `Sequence.enable_eom_mode()` and
+        `Sequence.disable_eom_mode()` methods.
+        A channel in EOM mode can only execute square pulses with a given
+        amplitude (`amp_on`) and detuning (`detuning_on`), which are
+        chosen at the moment the EOM mode is enabled. Furthermore, the
+        detuning when there is no pulse being played (`detuning_off`) is
+        restricted to a set of values that depends on `amp_on` and
+        `detuning_on`.
+        While in EOM mode, one can only add pulses of variable duration
+        (through `Sequence.add_eom_pulse()`) or delays.
+
+        Note:
+            Enabling the EOM mode will automatically enforce a buffer time from
+            the last pulse on the chose channel.
+
+        Args:
+            channel: The name of the channel to put in EOM mode.
+            amp_on: The amplitude of the EOM pulses (in rad/µs).
+            detuning_on: The detuning of the EOM pulses (in rad/µs).
+            detuning_off_choice: The criteria for choosing among the detuning
+                options for when there is no EOM pulse playing.
+
+                - ``'closest-to-zero'``: Picks the detuning value that's
+                    closest to zero.
+
+                - ``'highest'``: Picks the highest detuning value available.
+
+                - ``'lowest'``: Picks the lowest detuning value available.
```

Comment:
I must say, I'm not happy with this way of automatically selecting `detuning_off`.
I thought this three options would allow one to indirectly select each of the cases (both beams off, red on or blue on), but I have encountered an example where that is not the case.
What if, instead of having these protocols, we asked for an `optimal_detuning_off` value and then the selected `detuning_off` would be the closest to such a value?

Reply:
@madagra Do you have any thoughts on this?

Reply:
@HGSilveri I was thinking on the same line. An `optimal_detuning_off` looks a more flexible and clear option to me. However, what default value has to be assigned to the detuning if no `optimal_detuning_off` is passed?

Reply:
I think `optimal_detuning_off=0` is the only thing that makes sense because in regular operation the detuning is zero during delays.